Sources/APIClient/APIClient.swift

```diff
@@ -171,8 +172,10 @@ public final class APIClient {
     private func body<T: APIRequest>(for request: T) -> Data? {
         var bodyData: Data
         let encoder = JSONEncoder()
+        let dataToBeEncoded = request.body ?? request // if no .body is specified, encode the whole request
```

Comment:
So far I was solving this issue by writing a custom encoding method for each request object (see e.g. here https://github.com/ls1intum/artemis-ios/blob/46d758742d0faf9875967513f517dde4a1f30b0e/feature/Messages/Sources/Messages/Services/MessagesService/MessagesServiceImpl.swift#L258) , but this way it's even easier I think 👍